src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -148,10 +148,14 @@ private void OnFileChanged(FileSystemEventArgs e)
                 // will be offline
                 // when app_offline.htm is deleted, we trigger
                 // a restart to bring the host back online
+                // we also need to remove the debounce time here
+                // to avoid race condition causing the Host to
+                // not shutdown before a deployment starts
                 changeDescription = "File";
                 if (File.Exists(e.FullPath))
                 {
                     shutdown = true;
+                    _shutdown = Shutdown;
```

Comment:
Though this works, we may want to consider just calling shutdown and returning from here.
 1- Add an appropriate trace stating that `app_offline.htm` was detected and we're shutting down immediately
 2- That would avoid the tracing stating that shutdown was signaled
 3- Makes the intention clearer to others unfamiliar with the code.

Reply:
Ok, that makes sense. My initial thought process to not circuit out here was to keep the current tracing that happens below. I didn't know if it's generally ok to update log statements unless required, due to dependency some diagnostic tools may take. But I suppose, I could retain a similar logging here and Shutdown().